### ISSUE TITLE
game.getEventState is always 0

### ISSUE THREAD

**Issue by hendrik-weiler** [rpgboss/rpgboss#287]
https://github.com/rpgboss/rpgboss/blob/67f8ee8fcbf8089492ec5078ee34c947735640b6/common/src/main/resources/defaultrc/script/sys/eventEntity.js#L15

**Comment by tommycli** [comment]
game.getEventState is correct. Bug is in eventEntity.js.

**Comment by tommycli** [comment]
The "eventstate" global is never updated once it's set to something other than -1.

**Comment by hendrik-weiler** [comment]
How to check if the event state is changed, to get the comparison working?

**Comment by tommycli** [comment]
Actually your script is correct. Once the event state changes, it no longer calls the function recursively.

The actual game.getEventState call works fine. There's a unit test. (See EventSpec)

**Comment by hendrik-weiler** [comment]
It looks like even you say the script is correct.

I want to check in which state the event currently is, first tab = state 0,  second tab = state 1.

But this event state must have a different meaning.

So how to do that?

**Comment by tommycli** [comment]
Well I assumed you wanted to stop the loop once the event state changed.

If you want to check which state the event is currently in, the script is incorrect.

**Comment by hendrik-weiler** [comment]
Yes i want to stop the loop once the state changes from the state which the follow event selection change to another state which might have not that selection. Else the event will always follow the player.

So how i can archieve that?

**Comment by tommycli** [comment]
I think the Follow Player implementation should be in Scala for performance reasons. There shouldn't be any long-running loops in Javascript on separate threads in my opinion.

**Comment by hendrik-weiler** [comment]
Its ok. So how to do that?
